amp.js

```diff
@@ -61,6 +61,7 @@ createTask('release');
 createTask('serve');
 createTask('server-tests', 'serverTests');
 createTask('storybook');
+createTask('storybook-react');
```

Comment:
The `amp storybook` task has a flag called `--storybook_env` which takes `amp` or `preact` as modes. 

![image](https://user-images.githubusercontent.com/26553114/149990490-5d1e59e9-5be7-4567-b793-98a7a9112c71.png)

Curious about why we're adding a new task here instead of introducing a new mode to the existing flag. Are the code paths substantially different? If so, it may be possible to implement the guts in a separate file while retaining the original task name.

Reply:
storybook adds all `@storybook` packages, which creates conflicts between the amp, preact, and react versions of storybook. This was the only way I could figure out how to make them play nice with each other.

Reply:
Workaround, curious what y'all's thoughts are: What if we provide a `amp storybook-react-internal` task and then extend the primary `amp storybook` task to defer to it when `--storybook_env` includes `react`, this way we can centralize some of the logic/have a single standard entrypoint, even if technical challenges force it to be implemented separately?

Reply:
@rcebulko We can have a similar scheme without the necessity of subtasks.

The issue is a package install conflict in a case where Storybook does introspection. The workaround needs a separation of `package.json` files, a simpler consolidation proposal would look like:

```diff
- build-system/tasks/storybook-react/
+ build-system/tasks/storybook/react-env/
- build-system/tasks/storybook/package.json
+ build-system/tasks/storybook/amp-env/package.json 
+ build-system/tasks/storybook/preact-env/package.json 
```

It's probably okay if we consolidate after merging this PR, though.